lntest/itest/lnd_route_blinding.go

```diff
@@ -449,76 +449,63 @@ func (b *blindedForwardTest) createRouteToBlinded(paymentAmt int64,
 	require.Greater(b.t.t, len(resp.Routes), 0, "no routes")
 	require.Len(b.t.t, resp.Routes[0].Hops, 3, "unexpected route length")
 
-	return resp.Routes[0]
-}
-
-// sendBlindedPayment dispatches a payment to the route provided. The streaming
-// client for the send is returned with a cancel function that can be used to
-// terminate the stream.
-func (b *blindedForwardTest) sendBlindedPayment(route *lnrpc.Route) (
-	lnrpc.Lightning_SendToRouteClient, func()) {
-
-	hash := sha256.Sum256(b.preimage[:])
-
-	ctxt, cancel := context.WithCancel(b.ctx)
-	sendReq := &lnrpc.SendToRouteRequest{
-		PaymentHash: hash[:],
-		Route:       route,
+	// In order for the payment to go through the recipient (Dave) will
+	// need to have created an invoice.
+	// TODO(12/29/22): Do we need this function or would it be clearer
+	// to just create a single invoice by hand?
+	payReqs, rHashes, _, err := createPayReqs(
+		b.dave, lnwire.MilliSatoshi(paymentAmt).ToSatoshis(), 1,
+	)
+	if err != nil {
+		b.t.Fatalf("unable to create pay reqs: %v", err)
 	}
 
-	sendClient, err := b.net.Alice.SendToRoute(ctxt)
-	require.NoError(b.t.t, err, "send to route client")
-
-	err = sendClient.SendMsg(sendReq)
-	require.NoError(b.t.t, err, "send to route request")
-
-	return sendClient, cancel
-}
-
-// interceptFinalHop sets up a htlc interceptor on carol's incoming link to
-// intercept incoming htlcs.
-func (b *blindedForwardTest) interceptFinalHop() chan error {
-	interceptor, err := b.carol.RouterClient.HtlcInterceptor(b.ctx)
-	require.NoError(b.t.t, err, "interceptor")
-
-	hash := sha256.Sum256(b.preimage[:])
-
-	// Create an error channel to deliver any interceptor errors.
-	errChan := make(chan error)
-	go func() {
-		forward, err := interceptor.Recv()
+	// We'll need to provide the payment hash used for the invoice
+	// to callers so that their payment can correctly be associated
+	// with this invoice.
+	rHash := rHashes[0]
+
+	// Reconstruct the payment address.
+	var payAddr []byte
+	for _, payReq := range payReqs {
```

Comment:
Does invoice creation belong in this function? Rather than passing in `lnrpc.Invoice` and using the fields we need + letting the caller decide how to create the invoice.

Reply:
Definitely did not spend as long considering what should go where in the itest 😅